packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/ConvenienceMethodTemplateBase.java

```diff
@@ -114,9 +114,14 @@ protected void writeMethodImplementation(ClientMethod protocolMethod, ClientMeth
         // parameter transformation
         final ParameterTransformations transformations = convenienceMethod.getParameterTransformations();
         if (!transformations.isEmpty()) {
-            transformations.asStream()
-                .forEach(d -> writeParameterTransformation(d, convenienceMethod, protocolMethod, methodBlock,
-                    parametersMap));
+            transformations.asStream().forEach(d -> {
+                ClientMethodParameter requestBodyClientParameter
+                    = writeParameterTransformation(d, convenienceMethod, protocolMethod, methodBlock, parametersMap);
+                if (requestBodyClientParameter != null && !requestBodyClientParameter.isRequired()) {
+                    // protocol method parameter does not exist, set the parameter via RequestOptions
+                    methodBlock.line("requestOptions.setBody(" + requestBodyClientParameter.getName() + ");");
```

Comment:
iiuc, the bug was - we were not sending the body if the body is marked as optional (and in that case proxy interface method does not have a BinaryData argument), was that the case?

Reply:
Yes, the bug was that the body wasn't set to the `requestOptions`.

There is also a minor improvement that when no body need to be sent, code in `writeParameterTransformation` would not write the `BinaryData` for the body.